src/prefect/artifacts.py

```diff
@@ -124,7 +124,28 @@ async def create_table_artifact(
     Returns:
         - The table artifact ID.
     """
-    formatted_table = json.dumps(table) if isinstance(table, list) else table
+    if isinstance(table, dict):
+        for _, v in table.items():
+            if isinstance(v, list):
+                if any(isinstance(x, float) and x != x for x in v):
+                    raise ValueError(
+                        "`create_table_artifact` does not support NaN values in `table`"
+                        " argument."
+                    )
+        formatted_table = table
```

Comment:
I'm wondering if it would make sense to warn the user past a certain length. For instance, creating a table artifact like so: 

{"a": [1,2,...1000000], "b": [5]} 

would not be ideal.

Reply:
Don't we limit the size of the JSON payload?